src/modules/hyprland/workspaces.cpp

```diff
@@ -319,6 +330,12 @@ void add_or_remove_class(const Glib::RefPtr<Gtk::StyleContext> &context, bool co
 }
 
 void Workspace::update(const std::string &format, const std::string &icon) {
+  if (this->workspace_manager_.active_only() && !this->active()) {
```

Comment:
Once `urgent` workspaces are implemented in this module, we can simply add a check for this->urgent() here. Then it would work the same as in the `wlr/workspaces` module.